docs/src/userguide.md

```diff
@@ -194,6 +194,46 @@ The API details for AbstractTimestep object `t` are as follows:
 - useful functions for commonly used conditionals are `is_first(t)`,`is_last(t)`, `is_time(t, s)`, and `is_timestep(t, y)` as listed above
 - to access the index value of `t` as a `Number` representing the position in the time array, use `t.t`.  Users are encouraged to avoid this access, and instead use the options listed above or a separate counter variable. each time the function gets called. 
 
+Indexing into a variable or parameter's `time` dimension with an `Integer` is deprecated and will soon error. Instead, users should take advantage of the `TimestepIndex` and `TimestepValue` types. For examples we will refer back to our component definition above, and repeated below.
+```julia
+@defcomp MyComponentName begin
+  regions = Index()
+
+  A = Variable(index = [time])
+  B = Variable(index = [time, regions])
+
+  c = Parameter()
+  d = Parameter(index = [time])
+  e = Parameter(index = [time, regions])
+  f = Parameter(index = [regions])
+
+  function run_timestep(p, v, d, t)
+    v.A[t] = p.c + p.d[t]
+    for r in d.regions
+      v.B[t, r] = p.f[r] * p.e[t, r]
+    end
+  end
+
+end
+```
+`TimestepIndex` has one field, `index`, which refers to the absolute index in the parameter or variable array's `time` dimension. Thus, constructing a `TimestepIndex` is done by simply writing `TimestepIndex(index::Int)`. Looking back at our original component example
+one could modify the first line of `run_timestep` to always refer to the second first timestep of `p.d` with the following. One may index into the `time` dimension with a single `TimestepIndex`, or an `Array` of them.
```

Comment:
"always refer to the first timestep" (remove the accidental "second")
(also is there an unnecessary line break at the start of this line?)